tests/acceptance/features/lib/UsersPage.php

```diff
@@ -824,22 +836,28 @@ public function waitTillPageIsLoaded(
 		Session $session,
 		$timeout_msec = STANDARD_UI_WAIT_TIMEOUT_MILLISEC
 	) {
+		// There is always at least the "admin" user in the displayed list of users
+		// So wait for the user list to have at least 1 real user in it
 		$currentTime = \microtime(true);
 		$end = $currentTime + ($timeout_msec / 1000);
 		while ($currentTime <= $end) {
-			if ($this->findById($this->groupListId) !== null) {
-				break;
+			$userTrs = $this->findAll('xpath', $this->userTrXpath);
+			foreach ($userTrs as $userTr) {
+				$user = $userTr->find("css", ".name");
+				if ($this->getTrimmedText($user) !== '') {
+					// We have found a real user
+					// (note that there is a hidden empty "template" row)
+					break 2;
+				}
 			}
 			\usleep(STANDARD_SLEEP_TIME_MICROSEC);
 			$currentTime = \microtime(true);
 		}
 
 		if ($currentTime > $end) {
 			throw new \Exception(
-				__METHOD__ . " timeout waiting for users page to load"
+				__METHOD__ . " timeout waiting for user list to load on users page"
 			);
 		}
-
-		$this->waitForOutstandingAjaxCalls($session);
```

Comment:
The user list is the last thing loaded by AJAX. So the existence of a user in the list (checked above) means that the AJAX has finished. So calling `waitForOutstandingAjaxCalls()` here would be just a time-waster.

Note: previously I had trouble here, because various different loading of "resources" happens and `waitForOutstandingAjaxCalls()` would see a "moment" when no calls were outstanding. Then the AJAX to load the user list would happen - but the code here had already completed its "wait". A 2nd `waitForOutstandingAjaxCalls()` could have been pasted here - but that would be really ugly!